Formula/g/ghc.rb

```diff
@@ -7,12 +7,13 @@ class Ghc < Formula
   head "https://gitlab.haskell.org/ghc/ghc.git", branch: "master"
 
   livecheck do
-    url "https://www.haskell.org/ghc/download.html"
-    regex(/href=.*?download[._-]ghc[._-][^"' >]+?\.html[^>]*?>\s*?v?(\d+(?:\.\d+)+)\s*?</i)
+    url "https://www.haskell.org/ghc/"
+    regex(/href=.*?download_ghc_(\d+_\d+_\d+)\.html/i)
```

Comment:
```suggestion
    regex(/href=.*?download[_-]ghc[_-]v?(\d+(?:[._]\d+)+)\.html/i)
```

* There may be something to be said for using `(\d+(?:[._]\d+)+)` instead of `(\d+_\d+_\d+)`, as it would continue to work if the upstream URLs change to something like `download_ghc_1.2.3.html` in the future.
* Similarly, the prefix is `download_ghc_` but we may want to use `[_-]` to allow for `download-ghc-` in the future (e.g., they use hyphens in some other URLs like `/mailing-lists/`).